tests/api_test.py

```diff
@@ -2393,6 +2393,58 @@ def f(_):
     expected = jnp.arange(1) + 1
     self.assertAllClose(ans, expected)
 
+  def test_dot_precision_context_manager(self):
```

Comment:
A test or two specifically for `@` as matrix multiplication (perhaps via `operator.matmul`) might also be worthwhile.